src/app/cli/src/commands/system_api_server_run_command.rs

```diff
@@ -117,43 +115,52 @@ impl Command for APIServerRunCommand {
     async fn run(&mut self) -> Result<(), CLIError> {
         self.check_required_env_vars()?;
 
-        let access_token = self.get_access_token().await?;
-
-        // TODO: Cloning catalog is too expensive currently
-        let api_server = crate::explore::APIServer::new(
-            self.base_catalog.clone(),
-            &self.cli_catalog,
-            self.multi_tenant_workspace,
-            self.address,
-            self.port,
-        );
-
-        tracing::info!(
-            "API server is listening on: http://{}",
-            api_server.local_addr()
-        );
-
-        if self.output_config.is_tty
-            && self.output_config.verbosity_level == 0
-            && !self.output_config.quiet
-        {
-            eprintln!(
-                "{}\n  {}",
-                s("API server is listening on:").green().bold(),
-                s(format!("http://{}", api_server.local_addr())).bold(),
-            );
-            eprintln!("{}", s("Use Ctrl+C to stop the server").yellow());
-
-            if self.get_token {
-                eprintln!(
-                    "{} {}",
-                    s("JWT token:").green().bold(),
-                    s(access_token).dim()
+        // The borrow checker is happy if self is not explicitly used
+        let self_ref = &self;
+
+        run_transactional(
+            [&self.base_catalog, &self.cli_catalog],
+            move |[base_catalog, cli_catalog]| async move {
+                let access_token = self_ref.get_access_token(&base_catalog).await?;
+
+                let api_server = crate::explore::APIServer::new(
+                    base_catalog,
+                    &cli_catalog,
+                    self_ref.multi_tenant_workspace,
+                    self_ref.address,
+                    self_ref.port,
+                );
```

Comment:
This would create a single transaction for the entire run of the API server, so we will never save anything to the database, as API server never stops.

Reply:
yes, I realized my mistake